### PR TITLE
Set local confg dir for az cli so that multiple tasks can run in parallel

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
#8561

File: Tasks/AzureCLIV1/azureclitask.ts [hunk 1]
```diff
@@ -123,13 +128,29 @@ export class azureclitask {
...
         this.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
     }
 
+    private static setConfigDirectory(): void {
+        var configDirName: string = "c" + new Date().getTime(); // 'c' denotes config
+        if (tl.osType().match(/^Win/)) {
+            this.azCliConfigPath = path.join(process.env.USERPROFILE, ".azclitask", configDirName);
+        }
+        else {
+            this.azCliConfigPath = path.join(process.env.HOME, ".azclitask", configDirName);
```
Comment (hiyadav) [inline Tasks/AzureCLIV1/azureclitask.ts]: Are we sure that we always have permission to create new folder permissions in these paths ? 
Should we use some agent folder only or temp folder of respective os ?

Comment (hiyadav) [inline Tasks/AzureCLIV1/azureclitask.ts]: Are we sure that we always have permission to create new folder permissions in these paths ? 
Should we use some agent folder only or temp folder of respective os ?

**Comment by bishal-pdMSFT** [comment]
The paths are under user's profile: %USERPROFILE% for windows and /home/ for linux and the user would have read/write access to these paths. Infact az cli itself chooses a path in this location by default, hence the task would anyways be failing for such errors. Let me check if it falls back to some other location, then may be we also need to handle this.

**Comment by bishal-pdMSFT** [comment]
The paths are under user's profile: %USERPROFILE% for windows and /home/ for linux and the user would have read/write access to these paths. Infact az cli itself chooses a path in this location by default, hence the task would anyways be failing for such errors. Let me check if it falls back to some other location, then may be we also need to handle this.

**Comment by bishal-pdMSFT** [comment]
I checked az cli behavior and it fails if user does not have permission on %USERPROFILE%, so we are good with our changes. Task should fail if it does not have permission.

**Pull request merged**